config/configmodels/configmodels.go

```diff
@@ -106,19 +106,31 @@ func (dataType DataType) GetString() string {
 	case MetricsDataType:
 		return MetricsDataTypeStr
 	default:
-		panic("unknown data type")
+		return "custom"
+		// panic("unknown data type")
 	}
 }
 
 // Pipeline defines a single pipeline.
 type Pipeline struct {
+	TypeVal    Type     `mapstructure:"-"`
 	Name       string   `mapstructure:"-"`
 	InputType  DataType `mapstructure:"-"`
 	Receivers  []string `mapstructure:"receivers"`
 	Processors []string `mapstructure:"processors"`
 	Exporters  []string `mapstructure:"exporters"`
 }
 
+// Type gets the pipeline type.
+func (p *Pipeline) Type() Type {
+	return p.TypeVal
+}
+
+// SetType sets the pipeline type.
+func (p *Pipeline) SetType(typeStr Type) {
```

Comment:
Isn't this the data type?

Reply:
It's the type of the pipeline akin to type of a processor/exporter but it does overlap with our concept of datatype. We can model it either way.